pandas/core/dtypes/cast.py

```diff
@@ -350,9 +350,14 @@ def maybe_promote(dtype, fill_value=np.nan):
 
     # returns tuple of (dtype, fill_value)
     if issubclass(dtype.type, np.datetime64):
-        fill_value = tslibs.Timestamp(fill_value).value
+        fill_value = tslibs.Timestamp(fill_value).to_datetime64()
     elif issubclass(dtype.type, np.timedelta64):
-        fill_value = tslibs.Timedelta(fill_value).value
+        fv = tslibs.Timedelta(fill_value)
+        if fv is NaT:
+            # NaT has no `to_timedelta6` method
```

Comment:
really? should we add one instead of this?

Reply:
NaT is pretty specifically a datetime.  I think adding this there would muddy the waters

Reply:
ok